pkg/objstore/s3/s3.go

```diff
@@ -208,6 +216,17 @@ func validate(conf Config) error {
 	if conf.AccessKey != "" && conf.SecretKey == "" {
 		return errors.New("no s3 secret_key specified while access_key is present in config file; either both should be present in config or envvars/IAM should be used.")
 	}
+
+	// ensure that the prefix isn't too long
+	if conf.Prefix != "" && len(conf.Prefix) >= 1024 {
+		return errors.New("prefix is too long (limited by Amazon at 1024 bytes long)")
```

Comment:
I don't think this is a good idea, as S3 isn't only amazon it's also minio, ceph and other s3 compatible stores, so if other implementations support long prefixes, this would break those.

Reply:
OK, I can remove that.